userspace/libsinsp/filter.cpp

```diff
@@ -2040,30 +2044,64 @@ void sinsp_evttype_filter::add(string &name,
 			filters->push_back(wrap);
 		}
 	}
+
+	for(auto &tag: tags)
+	{
+		auto it = m_filter_by_tag.find(tag);
+
+		if(it == m_filter_by_tag.end())
+		{
+			m_filter_by_tag.emplace(std::make_pair(tag, std::list<filter_wrapper*>()));
+			it = m_filter_by_tag.find(tag);
+		}
+
+		it->second.push_back(wrap);
+	}
 }
 
-void sinsp_evttype_filter::enable(string &pattern, bool enabled)
+void sinsp_evttype_filter::enable(string &pattern, bool enabled, uint16_t ruleset)
 {
 	regex re(pattern);
 
-	for(auto val : m_evttype_filters)
+	for(auto &val : m_evttype_filters)
 	{
 		if (regex_match(val.first, re))
 		{
-			val.second->enabled = enabled;
+			if(val.second->enabled.size() < (size_t) (ruleset + 1))
+			{
+				val.second->enabled.resize(ruleset + 1);
+			}
+			val.second->enabled[ruleset] = enabled;
+		}
+	}
+}
+
+void sinsp_evttype_filter::enable_tags(set<string> &tags, bool enabled, uint16_t ruleset)
```

Comment:
That's not the behavior I would expect, actually. I think of this method as "find all rules matching these tags, and set their status to the value of enabled". It doesn't work on tags, it works on the rules that have a given tag. How about I clear up the comments to make that more explicit?